### ISSUE TITLE
[bug]: Unable to log in

### ISSUE THREAD

**Issue by mgmorcos** [hoppscotch/hoppscotch#3376]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When I press the login button, I expect to be logged in. This has only worked once. The rest of the time I am either taken back to the home screen and still not logged in, or I get an Unauthorized error from the API. I am using the latest version of the AIO image.

### Steps to reproduce

1. Press login button

### Environment

Production

### Version

Self-hosted

**Comment by mgmorcos** [comment]
Clearing cookies resolves the issue. Clearing cookies is necessary every time the server is restarted. Is there a way to invalidate sessions on the server side, so less technical users don't need to be asked to clear cookies?

**Comment by mgmorcos** [comment]
So far the best option I have is to change the session token before any session restart.

**Comment by mgmorcos** [comment]
So far the best option I have found is to change the session secret before every server restart. Then login works fine.

**Comment by balub** [comment]
@mgmorcos if the issue still persists, could you share the backend logs when the error happens?

**Comment by mgmorcos** [comment]
The logs are silent when login issues occur. The secret reset is not sufficient for all cases, sometimes clearing LocalStorage is also needed.

**Comment by dsole** [comment]
This could be related or duplicate of #3315

**Comment by mgmorcos** [comment]
My setup is on a single node without load balancing or clustering.

**Comment by dsole** [comment]
I was mainly thinking that even in a single node environment, when the server restarts it will be clearing the server's memory-based session store, so your cookie won't match a session anymore and needs to be cleared. 

So the problem of the session being memory-based rather than a persistent method was how they're potentially related.

**Comment by skywsp** [comment]
@dsole When I sign in with eamil, I got the same problem, and clearn cookies not working, what shoud I do... 

![image](https://github.com/hoppscotch/hoppscotch/assets/25943305/665f626f-66c6-460a-b237-00b53705c890)

**Comment by Bhupendrsingh** [comment]
> @dsole When I sign in with eamil, I got the same problem, and clearn cookies not working, what shoud I do...
> 
> ![image](https://user-images.githubusercontent.com/25943305/273434399-665f626f-66c6-460a-b237-00b53705c890.png)

Hi
i am also facing the same issue , are you able to resolve this

**Comment by SeptianENC** [comment]
I have same problem, when i click magic link the browser just refresh but i'm still not login.
does anyone have solution?

**Comment by pravindodia** [comment]
+1 for Email Auth Provider

**Comment by carsonbird** [comment]
So, is the impact of this bug that it is currently impossible to log in to the desktop app?

**Comment by fscaptain** [comment]
+1

**Comment by Teznew** [comment]
+1

**Comment by pravindodia** [comment]
Please change your enpoints for all from http to https and the login will work fine, its not mentioned in any document but I managed to login with it and it works 100%.

**Comment by carsonbird** [comment]
Hmm. Tried this, still didn't work. What authentication provider are you using (email, Google, GitHub, etc)?

**Comment by carsonbird** [comment]
@pravindodia nope, no proxy. Tried both methods with and without TLS, neither worked.

**Comment by pravindodia** [comment]
> @pravindodia nope, no proxy. Tried both methods with and without TLS, neither worked. 

Can you check the logs also see what error log you are getting ?

**Comment by carsonbird** [comment]
> Can you check the logs also see what error log you are getting ?

@pravindodia no errors in the console :p

**Comment by pravindodia** [comment]
> > Can you check the logs also see what error log you are getting ?
> 
> @pravindodia no errors in the console :p

@carsonbird What about browser network requests and responses?

**Comment by CharkiAbdelillah** [comment]
@pravindodia hey can you help me pls

**Comment by pravindodia** [comment]
> @pravindodia hey can you help me pls

What's your issue?

**Comment by mohijeet-changejar** [comment]
is there any workaround? i am still facing same issue it works fine with clearing cookies but not in first attempt, it shows 404 error page like its requesting content that does not exits but when i clear cache and reload page it logged me in.

**Comment by Hopp-magis** [comment]
Hello ! I need help too! I can't sign in i get this {"statusCode":500,"message":"Internal server error"}

**Comment by ge-infosec** [comment]
So what's the solution for this? I've this issue....

**Comment by abdul-musawwir002** [comment]
> Hello ! I need help too! I can't sign in i get this {"statusCode":500,"message":"Internal server error"}

Hello @Hopp-magis I have the same issue. Did your issue got resolved?

**Comment by nooodream** [comment]
i face the same issue，i use email and http to login, but when i click the email link, it not login in and let me send magic link again

**Comment by Hopp-magis** [comment]
let's check your env file
